src/precice/impl/DataContext.hpp

```diff
@@ -6,78 +6,170 @@
+   * 
+   * @return int ID of _fromData.
+   */
   int getFromDataID() const;
 
+  /**
+   * @brief Purpose unclear. See also https://github.com/precice/precice/issues/1156.
+   */
   void resetProvidedData();
 
+  /**
+   * @brief Resets _toData to zero. Used before mapping is performed.
+   */
   void resetToData();
 
+  /**
+   * @brief Get the ID of _toData. Used for performing the mapping outside of this class.
+   * 
+   * @return int ID of _toData.
+   */
   int getToDataID() const;
 
+  /**
+   * @brief Get the dimensions of _providedData.
+   * 
+   * @return int Dimensions of _providedData.
+   */
+  int getDataDimensions() const;
+
+  /**
+   * @brief Get the name of _mesh.
+   * 
+   * @return std::string Name of _mesh.
+   */
   std::string getMeshName() const;
 
+  /**
+   * @brief Get the ID of _mesh.
+   * 
+   * @return int ID of _mesh.
+   */
   int getMeshID() const;
 
   /**
-   * @brief links a read mapping and the mesh context the read mapping requires to this data context
-   *
-   * @param[in] mappingContext provides context of read mapping
-   * @param[in] fromMeshContext provides context of mesh this read mapping is mapping to (_toData)
+   * @brief Informs the user whether this DataContext has a _mappingContext.
+   * 
+   * @return True, if this DataContext is associated with a mapping. False, if not. 
    */
-  void configureForReadMapping(MappingContext mappingContext, MeshContext fromMeshContext);
+  bool hasMapping() const;
 
   /**
-   * @brief links a write mapping and the mesh context the write mapping requires to this data context
-   *
-   * @param[in] mappingContext provides context of write mapping
-   * @param[in] fromMeshContext provides context of mesh this write mapping is mapping from (_fromData)
+   * @brief Get the _mappingContext associated with this DataContext.
+   * 
+   * @return const MappingContext The _mappingContext of this DataContext.
    */
-  void configureForWriteMapping(MappingContext mappingContext, MeshContext toMeshContext);
-
-  bool hasMapping() const;
+  const MappingContext mappingContext() const;
 
+  /**
+   * @brief Informs the user whether this DataContext has a read mapping.
+   * 
+   * @return True, if DataContext has a read mapping.
+   */
   bool hasReadMapping() const;
 
+  /**
+   * @brief Informs the user whether this DataContext has a write mapping.
+   * 
+   * @return True, if DataContext has a write mapping.
+   */
   bool hasWriteMapping() const;
 
-  const MappingContext mappingContext() const;
+  /**
+   * @brief Links a MappingContext and the MeshContext required by the mapping to this DataContext.
+   * 
+   * A mapping maps the given data from or to _providedData (depending on whether it is a read or write mapping).
+   *
+   * @param[in] mappingContext Context of read mapping
+   * @param[in] meshContext Context of mesh this mapping is mapping from or to
+   */
+  virtual void configureMapping(MappingContext mappingContext, MeshContext meshContext) = 0;
```

Comment:
Please use pass by `const &` here

Reply:
Done in 555b9c2eae2e15e6ac1186ddcf6f0049a6c1dc9f.